content/kips/kip-29.md

```diff
@@ -0,0 +1,57 @@
+
+## Motivation
+
+1. Traders have expressed interest in copying trades of other accounts on Kwenta. Since futures market trades are all executed on-chain, traders have access to historical trading data for every other trading account.
+2. Traders want to promote others to copy trade them in exchange for a share of their profits.
+
+## Specification
+
+Terminology:
+- Copy trading account: A cross-margin (CM) enabled smart contract account which mirrors trades from another CM account 
+- Target account: The account being copied by a copy trading account
+
+The `MarginBase.sol` contract provides users with an interface to manage margin and futures positions across many markets in a single transaction. Keepers will use these functions to copy positions from the target account and execute trades on the copy trading account. To enable these functions the `MarginBase.sol` contract needs some new functionality:
+
+- Stores the address of the target account (`address targetAccount`.)
+- Stores an array of markets to be copied (`address[] copiedMarkets`.)
+- A function `rebalance` which matches the market exposure between the copy trading account and the target account when they are imbalanced.
+- Implements new access controls which prevent traders from submitting specific trades. Instead, these functions are delegated to a keeper account.
+- Implements a new `withdraw` function which closes positions proportionally and withdraws margin to the account owner.
+
+**Account Creation**
+
+Account creation follows a similar flow to cross margin, however users are required to specify two inputs at account creation time: `targetAccount` and `copiedMarkets`. Once an account is created, the account owner can deposit sUSD to the account. The rebalance function is called on deposit to copy positions between the copy trading account and the target account.
+
+**Rebalancing**
+
+When the `copiedTrader` account emits any `PositionModified` event, the keeper calls `rebalance` on the copy trading account. Rebalancing will open or close positions accordingly to give the copy trader the same market exposure as their target using the following logic:
+
+* `marginTotal = marginFree + marginMarket1 + ... + marginMarketN`
+* `freeMarginPct = marginFree / marginTotal`
+* `marginPctMarket1 = marginMarket1 / marginTotal` (calculate for all `copiedMarkets`)
+* `leverageMarket1 = notionalSizeMarket1 / marginMarket1` (calculate for all `copiedMarkets`)
+* Modify positions such that `freeMarginPct`, `marginPctMarketN`, and `leverageMarketN` are equal for all `copiedMarkets` between the copy trading account and the target account
+
+**Considerations and Risks**
+
+* Open interest caps could cause transactions to fail if the copy trading account trade sizes exceed caps during rebalancing
+* Dynamic fees could cause the copy trader to pay significant fees compared to their target since the transactions are executed sequentially
```

Comment:
We'd probably want to allow a user to configure some kind of control over how much deviation they are comfortable with from target traders fees / trade price, similar to what we have on limit orders / next-price. Also probably need to think about what happens if it does fail and the trade continues to move away outside of params it can execute on. Does this just become some feedback in the UI that the trade is now out of sync and the user has the ability to force rebalance back into sync or close.

Reply:
Fee rejection would be good to add here, great idea.